### PR TITLE
Add TryFrom &Vec<u8> to NodeId

### PR DESCRIPTION
**Pull request by maaktweluit**
Very handy conversion used a lot in ya-relay

https://github.com/golemfactory/ya-relay/pull/76

### CODE REVIEW

File: model/src/node_id.rs [hunk 1]
```diff
@@ -81,6 +82,16 @@ impl From<[u8; 20]> for NodeId {
     }
 }
 
+impl TryFrom<&Vec<u8>> for NodeId {
+    type Error = &'static str;
+    fn try_from(inner: &Vec<u8>) -> Result<Self, Self::Error> {
+        if inner.len() != 20 {
```
Comment (nieznanysprawiciel) [inline model/src/node_id.rs:88]:
```suggestion
        if inner.len() != NodeId::default().len() {
```

File: model/src/node_id.rs [hunk 2]
```diff
@@ -81,6 +82,16 @@ impl From<[u8; 20]> for NodeId {
     }
 }
 
+impl TryFrom<&Vec<u8>> for NodeId {
+    type Error = &'static str;
```
Comment (nieznanysprawiciel) [inline model/src/node_id.rs:86]: Returning string is not a good idea as I commented in related `ya-relay` PR

File: model/src/node_id.rs [hunk 3]
```diff
@@ -21,6 +21,13 @@ impl ParseError {
     }
 }
 
+#[derive(Clone, Debug, thiserror::Error, PartialEq, Serialize, Deserialize)]
+#[error("NodeId `{input:?}` parsing error: {msg}")]
```
Comment (nieznanysprawiciel) [inline model/src/node_id.rs:27]: This will be ugly printed :(

Comment (maaktweluit) [reply model/src/node_id.rs:27]: Vec<u8> does not support `:x` printing :(

**Review by nieznanysprawiciel — DISMISSED** [review]

**Review by nieznanysprawiciel — APPROVED** [review]

**Pull request merged**